From fa37195d8378ac958d1f9bd884b47bd73730040e Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Thu, 27 Dec 2018 09:58:07 -0800
Subject: [PATCH] imap-send: Fix compilation without deprecated OpenSSL APIs

Initialization in OpenSSL has been deprecated in version 1.1. This makes
compilation fail when deprecated APIs for OpenSSL are compile-time
disabled.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 imap-send.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/imap-send.c
+++ b/imap-send.c
@@ -259,8 +259,10 @@ static int ssl_socket_connect(struct ima
 	int ret;
 	X509 *cert;
 
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
 	SSL_library_init();
 	SSL_load_error_strings();
+#endif
 
 	meth = SSLv23_method();
 	if (!meth) {
